packages/ref-metrics-pmp/src/cmip_ref_metrics_pmp/annual_cycle.py

```diff
@@ -0,0 +1,228 @@
+
+from loguru import logger
+
+from cmip_ref_core.datasets import FacetFilter, SourceDatasetType
+from cmip_ref_core.metrics import (
+    CommandLineMetric,
+    DataRequirement,
+    MetricExecutionDefinition,
+    MetricExecutionResult,
+)
+from cmip_ref_metrics_pmp.pmp_driver import build_pmp_command, process_json_result
+
+
+class AnnualCycle(CommandLineMetric):
+    """
+    Calculate the annual cycle for a dataset
+    """
+
+    def __init__(self) -> None:
+        self.name = "PMP Annual Cycle"
+        self.slug = "pmp-annual-cycle"
+        self.data_requirements = (
+            DataRequirement(
+                source_type=SourceDatasetType.PMPClimatology,
+                filters=(
+                    FacetFilter(
+                        facets={"source_id": ("GPCP-Monthly-3-2", "ERA-5"), "variable_id": ("pr", "ts")}
+                    ),
+                ),
+                group_by=("variable_id", "source_id"),
+            ),
+            DataRequirement(
+                source_type=SourceDatasetType.CMIP6,
+                filters=(
+                    FacetFilter(
+                        facets={
+                            "frequency": "mon",
+                            "experiment_id": ("amip", "historical", "hist-GHG", "piControl"),
+                            "variable_id": ("pr", "ts"),
+                        }
+                    ),
+                ),
+                group_by=("variable_id", "source_id", "experiment_id", "variant_label", "member_id"),
+            ),
+        )
+
+        self.parameter_file_1 = "pmp_param_annualcycle_1-clims.py"
+        self.parameter_file_2 = "pmp_param_annualcycle_2-metrics.py"
+
+    def build_cmd(self, definition: MetricExecutionDefinition) -> Iterable[str]:
+        """
+        Build the command to run the metric
+
+        Parameters
+        ----------
+        definition
+            Definition of the metric execution
+
+        Returns
+        -------
+            Command arguments to execute in the PMP environment
+        """
+        # Pass the parameters using **kwargs
+        raise NotImplementedError("Function not required")
+
+    def build_cmds(self, definition: MetricExecutionDefinition) -> list[list[str]]:
+        """
+        Build the command to run the metric
+
+        Parameters
+        ----------
+        definition
+            Definition of the metric execution
+
+        Returns
+        -------
+            Command arguments to execute in the PMP environment
+        """
+        input_datasets = definition.metric_dataset[SourceDatasetType.CMIP6]
+        source_id = input_datasets["source_id"].unique()[0]
+        experiment_id = input_datasets["experiment_id"].unique()[0]
+        member_id = input_datasets["member_id"].unique()[0]
+        variable_id = input_datasets["variable_id"].unique()[0]
+        model_files = input_datasets.path.to_list()[0]  # Limits to one file, needs to be fixed in the future
```

Comment:
Perhaps this should raise an exeception if more than one file is passed like the other metric

Reply:
Great point! Enabled multi-file-cases by commits https://github.com/Climate-REF/climate-ref/pull/221/commits/ab00e1093cf8f731b9e317cfdd9f4799e84269e9 and https://github.com/Climate-REF/climate-ref/pull/221/commits/6f2cd30f4f05a2c0c9263d0ac403d293f36722aa.